src/app/qgisapp.cpp

```diff
@@ -12402,6 +12428,47 @@ void QgisApp::writeDockWidgetSettings( QDockWidget *dockWidget, QDomElement &ele
   elem.setAttribute( QStringLiteral( "area" ), dockWidgetArea( dockWidget ) );
 }
 
+bool QgisApp::askForDatumTransform( QgsCoordinateReferenceSystem sourceCrs, QgsCoordinateReferenceSystem destinationCrs )
```

Comment:
Is it safe to call this out of the main thread?

I think it would be good to add a `Q_ASSERT` on top to get an immediate feedback for unsafe handling in dev environments.

Another thought is to add a non-modal version (especially for out of main-thread events), where right now a default config is used and the user is informed about a missing configuration for these CRS'es with a button/link to fix this missing definition later on.

Reply:
added Q_ASSERT

there is a similar case handled for project CRS changed which triggers `QgisApp::projectCrsChanged`